src/hostname/hostnamed.c

```diff
@@ -601,6 +613,25 @@ static int property_get_hardware_model(
         return property_get_hardware_property(reply, userdata, PROP_HARDWARE_MODEL, get_hardware_model);
 }
 
+static int property_get_firmware_version(
+                sd_bus *bus,
+                const char *path,
+                const char *interface,
+                const char *property,
+                sd_bus_message *reply,
+                void *userdata,
+                sd_bus_error *error) {
+
+        _cleanup_free_ char *firmware_version = NULL;
+        int r;
+
+        r = get_firmware_version(&firmware_version);
+        if (r < 0)
+                return r;
```

Comment:
Ah, try to voidify the call. Some devices may not have the dmi devices.

Reply:
ok. i thought it's related to this https://github.com/systemd/systemd/issues/4753. Will try to voidify it.

Reply:
Will squash commits if ci passes

Reply:
done